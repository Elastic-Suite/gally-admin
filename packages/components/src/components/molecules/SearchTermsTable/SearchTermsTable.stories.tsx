import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import SearchTermsTableComponent from './SearchTermsTable'
import { useTranslation } from 'next-i18next'

export default {
  title: 'Molecules/SearchTermsTable',
  component: SearchTermsTableComponent,
} as ComponentMeta<typeof SearchTermsTableComponent>

function WithTranslation({
  data,
  title,
}: React.ComponentProps<typeof SearchTermsTableComponent>): JSX.Element {
  const { t } = useTranslation('searchUsage')
  const count = data.length
  return (
    <SearchTermsTableComponent
      title={title || t('resultsSearchItems', { count })!}
      data={data}
    />
  )
}
const Template: ComponentStory<typeof SearchTermsTableComponent> = (args) => {
  return <WithTranslation {...args} />
}

export const DefaultCase = Template.bind({})

const data = [
  {
    term: 'Colle cleopatre pour slime',
    session: 6,
  },
  {
    term: 'Classeur pokemon',
    session: 4,
  },
  {
    term: 'Grimoire vierge',
    session: 1,
  },
  {
    term: 'Chromecast',
    session: 6,
  },
]

DefaultCase.args = {
  data,
}
