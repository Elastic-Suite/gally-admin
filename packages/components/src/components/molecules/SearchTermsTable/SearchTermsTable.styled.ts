import { styled } from '@mui/material'

export const Title = styled('h3')(({ theme }) => ({
  fontSize: theme.spacing(2.5),
  fontWeight: 600,
  color: theme.palette.colors.neutral[900],
  whiteSpace: 'normal',
  fontFamily: 'var(--gally-font)',
}))
