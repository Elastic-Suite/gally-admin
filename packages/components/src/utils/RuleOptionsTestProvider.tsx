import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
