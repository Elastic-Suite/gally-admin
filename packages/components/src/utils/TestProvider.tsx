import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
