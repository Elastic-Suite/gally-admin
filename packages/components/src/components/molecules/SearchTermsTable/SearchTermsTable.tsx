import React from 'react'
import ReadOnlyTable from '../../atoms/ReadOnlyTable/ReadOnlyTable'
import { IRow } from '../../atoms/ReadOnlyTable/ReadOnlyTable.styled'
import { Title } from './SearchTermsTable.styled'
import { useTranslation } from 'next-i18next'

interface ISearchTermsTableProps {
  title?: string
  data: {
    term: string
    session: number
  }[]
}

function SearchTermsTable({
  title,
  data,
}: ISearchTermsTableProps): JSX.Element {
  const { t } = useTranslation('searchUsage')

  const header = {
    cells: [
      {
        id: 'terms',
        value: t('Terms'),
        style: {
          width: '50%',
        },
      },
      {
        id: 'sessions',
        value: t('Sessions'),
        style: {
          width: '50%',
        },
      },
    ],
  }
  const body: IRow[] = data
    .sort((a, b) => b.session - a.session)
    .map(({ term, session }, index) => ({
      id: `row-${index}`,
      cells: [
        {
          id: `term-${index}`,
          value: term,
        },
        {
          id: `session-${index}`,
          value: session,
        },
      ],
    }))

  return (
    <div style={{ width: '100%' }}>
      {Boolean(title) && <Title>{title}</Title>}
      <ReadOnlyTable header={header} body={body} fullWidth />
    </div>
  )
}

export default SearchTermsTable
