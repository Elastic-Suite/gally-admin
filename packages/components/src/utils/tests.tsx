import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
