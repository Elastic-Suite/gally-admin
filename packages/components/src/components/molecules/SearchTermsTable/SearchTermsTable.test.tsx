import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import SearchTermsTable from './SearchTermsTable'

describe('SearchTermsTable', () => {
  it('SearchTermsTable match snapshot', () => {
    const { container } = renderWithProviders(
      <SearchTermsTable
        data={[
          { term: 'Test 1', session: 1 },
          { term: 'Test 2', session: 2 },
          { term: 'Test 3', session: 3 },
        ]}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
