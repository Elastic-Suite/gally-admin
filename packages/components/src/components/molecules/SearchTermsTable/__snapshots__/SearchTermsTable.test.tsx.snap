// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTermsTable SearchTermsTable match snapshot 1`] = `
<div>
  <div
    style="width: 100%;"
  >
    <table
      class="css-1k86jay"
      data-testid="readOnlyTable"
    >
      <thead>
        <tr
          class=""
          data-testid="readOnlyTableRow"
        >
          <td
            style="width: 50%;"
          >
            Terms
          </td>
          <td
            style="width: 50%;"
          >
            Sessions
          </td>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          data-testid="readOnlyTableRow"
        >
          <td>
            Test 3
          </td>
          <td>
            3
          </td>
        </tr>
        <tr
          class=""
          data-testid="readOnlyTableRow"
        >
          <td>
            Test 2
          </td>
          <td>
            2
          </td>
        </tr>
        <tr
          class=""
          data-testid="readOnlyTableRow"
        >
          <td>
            Test 1
          </td>
          <td>
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
